draft-ietf-jsonpath-base.md

```diff
@@ -1115,54 +1115,75 @@ basic-expr        = exist-expr /
                     paren-expr /
                     relation-expr
 exist-expr        = [neg-op S] singular-path          ; path existence or non-existence
+~~~~
+
+Paths in filter expressions are Singular Paths, each of which selects at most one node.
+
+~~~~ abnf
 singular-path     = rel-singular-path / abs-singular-path
 rel-singular-path = "@" *(S (dot-selector / index-selector))
 abs-singular-path = root-selector *(S (dot-selector / index-selector))
+~~~~
+
+Parentheses can be used with `boolean-expr` for grouping. So filter selection syntax in the original proposal `[?(<expr>)]` is naturally contained in the current lean syntax `[?<expr>]` as a special case.
+
+~~~~ abnf
 paren-expr        = [neg-op S] "(" S boolean-expr S ")" ; parenthesized expression
 neg-op            = "!"                               ; not operator
 
 relation-expr = comp-expr /                           ; comparison test
                 regex-expr                            ; regular expression test
+~~~~
+
+Comparisons are restricted to Singular Path values and primitive values (such as number, string, `true`, `false`, `null`).
+
+Comparisons with complex values will fail, i.e. no selection occurs.
+<!-- issue: comparison with structured value -->
 
+Data types are not implicitly converted in comparisons.
+So `"13 == '13'"` selects no node.
+
+A member or element value by itself in a Boolean context is
+interpreted as `false` only if it does not exist.
+Otherwise it is interpreted as `true`.
+To be more specific about the actual value, explicit comparisons are necessary. This existence test — as an exception to the general rule — also works with structured values.
+
+~~~~ abnf
 comp-expr    = comparable S comp-op S comparable
 comparable   = number / string-literal /              ; primitive ...
                true / false / null /                  ; values only
                singular-path                          ; Singular Path value
 comp-op      = "==" / "!=" /                          ; comparison ...
                "<"  / ">"  /                          ; operators
                "<=" / ">="
+~~~~
+
+Alphabetic characters in ABNF are case-insensitive, so "e" can be either "e" or "E".
```

Comment:
With RFC 7405 you can say %s and get case-sensitive matching.  But I think this is OK?

Reply:
I think it's ok. We don't require a normalized form of `number`. For example, I think the following all have the same meaning in this context: `100`, `100.0`, `100.00`, etc., `1e2`, `1.0e2`, `1.00e2`, etc., `1e02`, `1e002`, etc., `1000e-1`, `10000e-2`, etc. (and then the forms using `E` instead of `e`).